addons/sale/models/sale_order.py

```diff
@@ -1051,11 +1051,18 @@ def _find_mail_template(self):
         :rtype: record of `mail.template` or `None` if not found
         """
         self.ensure_one()
-        if self.env.context.get('proforma') or self.state != 'sale':
-            return self.env.ref('sale.email_template_edi_sale', raise_if_not_found=False)
-        else:
+        template_ref = 'sale.email_template_edi_sale'
+
+        if self.env.context.get('proforma'):
+            template_ref = 'sale.email_template_edi_sale_proforma'
+
+        template = self.env.ref(template_ref, raise_if_not_found=False)
+
+        if not template and self.state == 'sale':
             return self._get_confirmation_template()
 
+        return template
+
```

Comment:
You're modifying the behavior here, we won't ever enter `_get_confirmation_template` anymore :eyes:

Reply:
True :eyes: will change that soon ahah